src/services/graphql.ts

```diff
@@ -0,0 +1,199 @@
+import gql from 'graphql-tag';
+
+// tslint:disable:max-line-length
+
+function getLineQuery() {
+    return (
+        gql`query getLineByLintunnus ($lineId: String!) {
+            linjaByLintunnus(lintunnus:$lineId) {
+                ${lineQueryFields}
+                reittisByLintunnus(first: 1, orderBy: REIVIIMPVM_DESC) {
+                    nodes {
+                        reinimi
+                        reiviimpvm
+                    }
+                }
+            }
+        }`
+    );
+}
+
+function getAllLinesQuery() {
+    return (
+        gql`{
+            allLinjas {
+                nodes {
+                    ${lineQueryFields}
+                    reittisByLintunnus(orderBy: REIVIIMPVM_DESC) {
+                        nodes {
+                            reinimi
+                            reitunnus
+                            reiviimpvm
+                        }
+                    }
+                }
+            }
+        }`
+    );
+}
+
+function getRouteQuery() {
+    return (
+        gql`query getLineDetails($routeId: String!) {
+            route: reittiByReitunnus(reitunnus: $routeId) {
+                ${routeQueryFields}
+                reitinsuuntasByReitunnus{
+                    nodes {
+                        ${routePathQueryFields}
+                        reitinlinkkisByReitunnusAndSuuvoimastAndSuusuunta {
+                            nodes {
+                                ${routePathLinkQueryFields}
+                                solmuByLnkalkusolmu {
+                                    ${startNodeQueryFields}
+                                }
+                                solmuByLnkloppusolmu {
+                                    ${endNodeQueryFields}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }`
+    );
+}
+
+function getRoutePathQuery() {
+    return (
+        gql`query getRoutePath($routeId: String!, $startDate: Datetime!, $direction: String!) {
+            routePath: reitinsuuntaByReitunnusAndSuuvoimastAndSuusuunta(reitunnus: $routeId, suuvoimast: $startDate, suusuunta: $direction) {
+                ${routePathQueryFields}
+                reittiByReitunnus {
+                    lintunnus
+                }
+                reitinlinkkisByReitunnusAndSuuvoimastAndSuusuunta {
```

Comment:
This is exactly identical to the code used on row 48, in the function above. Could we reuse this code?

Reply:
const reitinlinkkiQuery = ...